ScanCommand/src/main/java/io/ballerina/scan/ScanUtils.java

```diff
@@ -0,0 +1,95 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.scan;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import io.ballerina.projects.Project;
+import io.ballerina.projects.internal.model.Target;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintStream;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import static io.ballerina.scan.ScanToolConstants.RESULTS_JSON_FILE;
+
+class ScanUtils {
+    private ScanUtils() {
+    }
+
+    static void printToConsole(List<Issue> issues, PrintStream outputStream) {
+        String jsonOutput = convertIssuesToJsonString(issues);
+        outputStream.println();
+        outputStream.println(jsonOutput);
+    }
+
+    private static String convertIssuesToJsonString(List<Issue> issues) {
+        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        JsonArray issuesAsJson = gson.toJsonTree(issues).getAsJsonArray();
+        return gson.toJson(issuesAsJson);
+    }
+
+    static Path saveToDirectory(List<Issue> issues, Project project, String directoryName) {
+        Target target = getTargetPath(project, directoryName);
+
+        Path jsonFilePath;
+        try {
+            Path reportPath = target.getReportPath();
+            String jsonOutput = convertIssuesToJsonString(issues);
+            File jsonFile = new File(reportPath.resolve(RESULTS_JSON_FILE).toString());
+
+            try (FileOutputStream fileOutputStream = new FileOutputStream(jsonFile)) {
+                try (Writer writer = new OutputStreamWriter(fileOutputStream, StandardCharsets.UTF_8)) {
+                    writer.write(new String(jsonOutput.getBytes(StandardCharsets.UTF_8), StandardCharsets.UTF_8));
+                    writer.close();
```

Comment:
Closing the writer explicitly is unnecessary because it's automatically closed when exiting the try-with-resources block.

```suggestion
```